test/range.civet

```diff
@@ -56,15 +56,73 @@ describe "range", ->
     ---
     [0...256]
     ---
-    ((s) => Array.from({length: 256 - s}, (_, i) => s + i))(0)
+    var range: (start: number, end: number) => number[] = (start, end) => {
+      const length = end - start;
+      if (length <= 0) return [];
+      const arr = Array(length);
+      for (let i = 0; i < length; ++i) {
+        arr[i] = i + start;
+      }
+      return arr;
+    };
+    range(0,256)
   """
 
+  closedAscendingArr := [0..100]
+  halfClosedAscendingArr := [0...100]
+  openAscendingArr := [1..<100]
+
+  evalsTo '[0..100]', closedAscendingArr
+  evalsTo '[0..<=100]', closedAscendingArr
+  evalsTo '[0<=..<=100]', closedAscendingArr
+  evalsTo '[0...100]', halfClosedAscendingArr
+  evalsTo '[0..<100]', halfClosedAscendingArr
+  evalsTo '[0<=..<100]', halfClosedAscendingArr
+  evalsTo '[0<..<100]', openAscendingArr
+
+  closedDescendingArr: number[] := [100>=..>=0]
+  halfClosedDescendingArr: number[] := [100>=..>0]
+  openDescendingArr: number[] := [99>=..>0]
```

Comment:
We could even call `reverse` on the ascending arrays here...

Reply:
Not for the half-closed one -- ascending is `[0, 100)` and descending is `(0, 100]`. I'll do that for the other two though.